arangod/Cluster/HeartbeatThread.cpp

```diff
@@ -1088,6 +1086,7 @@ bool HeartbeatThread::init() {
   if (ServerState::instance()->isClusterRole() && !sendServerState()) {
     return false;
   }
+  _maintenanceThread = std::make_unique<HeartbeatBackgroundJobThread>(_server, this);
```

Comment:
As far as I can see, this starts the maintenanceThread now in all server modes (i.e. coordinator and DB server).
But AFAIK only the DB server should have this thread. So I think initializing in the original place was better.

Reply:
The problem is that this object is accessed by some other thread that is already running at the time that old code would initialize it. So the intention was to initialize it _before_ this other thread is started. If this thread is only needed in the agency we can add an according if.